src/Kdyby/Doctrine/Proxy/ProxyAutoloaderListener.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * This file is part of the Kdyby (http://www.kdyby.org)
+ *
+ * Copyright (c) 2008 Filip Procházka (filip@prochazka.su)
+ *
+ * For the full copyright and license information, please view the file license.txt that was distributed with this source code.
+ */
+
+namespace Kdyby\Doctrine\Proxy;
+
+use Kdyby;
+use Nette;
+
+/**
+ * @author Jáchym Toušek <enumag@gmail.com>
+ */
+class ProxyAutoloaderListener extends Nette\Object implements Kdyby\Events\Subscriber
+{
+
+	/** @var array */
+	private $proxyAutoLoaders;
+
+
+
+	public function __construct(array $proxyAutoLoaders)
+	{
+		$this->proxyAutoLoaders = $proxyAutoLoaders;
```

Comment:
It's not like you wanna unregister them anyway. And they are an array just because it was convinient, you don't really have multiple proxy dirs in production unless you have multiple entitymanager configurations and even then, it's still more likely you'll have only one.

I didn't mean to move the array of directories to the proxyautoloader, but rather make each of those autoloaders a subscriber.